desktop/cmp/contextmenu/StoreContextMenu.js

```diff
@@ -42,7 +42,9 @@ export class StoreContextMenu {
     constructor({items, gridModel}) {
         this.gridModel = gridModel;
         this.items = flatten(items.map(it => {
-            return isString(it) ? this.parseToken(it) : new RecordAction(it);
+            if (it instanceof RecordAction) return it;
```

Comment:
This is dangerous since the RecordActions are now used as the backing state for multiple UI components, which can/will be on the screen simultaneously. If we do this then we will run into issues if users are passing the same RecordAction instance to the context menu and action bar.

I think we do want to always make a clone of the RecordActions which are passed to the StoreContextMenu and RecordActionBar. Agree that depending on the RecordAction ctor destructuring for cloning the way I was is also dangerous and likely to break in the future, should we instead do a lodash `cloneDeep` here?